packages/vercel/src/index.ts

```diff
@@ -0,0 +1,619 @@
+							for (const route of routes) {
+								const src = route.pattern.source;
+								const dest = src.startsWith('^\\/_image') ? NODE_PATH : ISR_PATH;
+								if (!route.prerender) routeDefinitions.push({ src, dest });
+							}
+						} else {
+							await builder.buildServerlessFolder(entryFile, NODE_PATH, _config.root);
+							const dest = _middlewareEntryPoint ? MIDDLEWARE_PATH : NODE_PATH;
+							for (const route of routes) {
+								if (!route.prerender) routeDefinitions.push({ src: route.pattern.source, dest });
+							}
+						}
+					}
+					if (_middlewareEntryPoint) {
+						await builder.buildMiddlewareFolder(
+							_middlewareEntryPoint,
+							MIDDLEWARE_PATH,
+							middlewareSecret
+						);
+					}
+				}
+				const fourOhFourRoute = routes.find((route) => route.pathname === '/404');
+				const destination =
+					buildOutput === 'server'
+						? new URL('./config.json', _config.outDir)
+						: new URL('./.vercel/output/config.json', _config.root);
+				const finalRoutes = [
+					...getRedirects(routes, _config),
+					{
+						src: `^/${_config.build.assets}/(.*)$`,
+						headers: { 'cache-control': 'public, max-age=31536000, immutable' },
+						continue: true,
+					},
+					{ handle: 'filesystem' },
+				];
+				if (buildOutput === 'server') {
+					finalRoutes.push(...routeDefinitions);
+				}
+
+				if (fourOhFourRoute) {
+					if (buildOutput === 'server') {
+						finalRoutes.push({
+							src: '/.*',
+							dest: fourOhFourRoute.prerender
+								? '/404.html'
+								: _middlewareEntryPoint
+									? MIDDLEWARE_PATH
+									: NODE_PATH,
+							status: 404,
+						});
+					} else {
+						finalRoutes.push({
+							src: '/.*',
+							dest: '/404.html',
+							status: 404,
+						});
+					}
+				}
+
+				let images: VercelImageConfig | undefined;
+				if (imageService || imagesConfig) {
+					if (imagesConfig) {
+						images = {
+							...imagesConfig,
+							domains: [...imagesConfig.domains, ..._config.image.domains],
+							remotePatterns: [...(imagesConfig.remotePatterns ?? [])],
+						};
+						const remotePatterns = _config.image.remotePatterns;
+						for (const pattern of remotePatterns) {
+							if (isAcceptedPattern(pattern)) {
+								images.remotePatterns?.push(pattern);
+							}
+						}
+					} else {
+						images = getDefaultImageConfig(_config.image);
+					}
+				}
+
+				// Output configuration
+				// https://vercel.com/docs/build-output-api/v3#build-output-configuration
+				await writeJson(destination, {
+					version: 3,
+					routes: finalRoutes,
+					images,
+				});
+
+				// Remove temporary folder
+				if (buildOutput === 'server') {
+					await removeDir(_buildTempFolder);
+				}
+			},
+		},
+	};
+}
+
+function isAcceptedPattern(pattern: any): pattern is RemotePattern {
+	if (!pattern?.hostname) {
+		return false;
+	}
+	if (pattern?.protocol === 'http' || pattern?.protocol === 'https') {
+		return true;
+	}
+	return false;
+}
```

Comment:
```suggestion
function isAcceptedPattern(pattern: any): pattern is RemotePattern {
	if (pattern == null) {
		return false;
	}
	if (!pattern.hostname) {
		return false;
	}
	if (pattern.protocol && (pattern.protocol !== 'http' || pattern.protocol !== 'https')) {
		return false;
	}
	return true;
}
```

>  Also, I think this check still requires RemotePattern to have the protocol field, however the types showed that it's optional

My code above should address this as mentioned before. Can you please double check the logic 🙏